src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -117,40 +118,56 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
  * using a temporary default tsconfig.json file.
  */
 async function compileTypeScript(config: NodejsDebugConfiguration): Promise<void> {
-    if (config.invokeTarget.target === 'code') {
-        const tscResponse = await new ChildProcess(true, 'tsc', undefined, '-v').run()
-        if (tscResponse.exitCode !== 0 || !tscResponse.stdout.startsWith('Version')) {
-            throw new Error('TypeScript compiler "tsc" not found.')
-        }
-        const samBuildOutputAppRoot = path.join(
-            config.baseBuildDir!,
-            'output',
-            path.parse(config.invokeTarget.projectRoot).name
-        )
-        const tsconfigPath = path.join(samBuildOutputAppRoot, 'tsconfig.json')
-        if (
-            (await readdir(config.codeRoot)).includes('tsconfig.json') ||
-            (await hasFileWithSuffix(config.codeRoot, '.ts', '**/node_modules/**'))
-        ) {
-            //  This default config is a modified version from the AWS Toolkit for JetBrain's tsconfig file. https://github.com/aws/aws-toolkit-jetbrains/blob/911c54252d6a4271ee6cacf0ea1023506c4b504a/jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsLambdaBuilder.kt#L60
-            const defaultTsconfig = {
-                compilerOptions: {
-                    target: 'es6',
-                    module: 'commonjs',
-                    typeRoots: ['node_modules/@types'],
-                    types: ['node'],
-                    rootDir: '.',
-                    inlineSourceMap: true,
-                },
-            }
-            try {
-                writeFileSync(tsconfigPath, JSON.stringify(defaultTsconfig, undefined, 4))
-                getLogger('channel').info('Compiling TypeScript')
-                await new ChildProcess(true, 'tsc', undefined, '--project', samBuildOutputAppRoot).run()
-            } catch (error) {
-                getLogger('channel').error(`Compile Error: ${error}`)
-                throw Error('Failed to compile typescript Lambda')
-            }
-        }
+    if (!config.baseBuildDir) {
+        throw Error('invalid state: config.baseBuildDir was not set')
+    }
+    if (config.invokeTarget.target !== 'code') {
+        return
+    }
+
+    const isTypescriptProject =
+        (await readdir(config.codeRoot)).includes('tsconfig.json') ||
```

Comment:
reading the whole dir just to check one file can be probably optimised to just one `existsSync`?